token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -1,19 +1,183 @@
+    *mint.init_extension::<ConfidentialTransferAuditor>()? = *auditor;
+
+    Ok(())
+}
+
+/// Processes a [ConfigureOmnibusAccount] instruction.
+fn process_configure_omnibus_account(accounts: &[AccountInfo]) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let funder_info = next_account_info(account_info_iter)?;
+    let mint_info = next_account_info(account_info_iter)?;
+    let omnibus_info = next_account_info(account_info_iter)?;
+    let system_program_info = next_account_info(account_info_iter)?;
+
+    check_program_account(mint_info.owner)?;
+    let _mint = StateWithExtensions::<Mint>::unpack(&mint_info.data.borrow())?;
+
+    let (omnibus_token_address, omnibus_token_bump_seed) =
+        get_omnibus_token_address_with_seed(mint_info.key);
+
+    if omnibus_token_address != *omnibus_info.key {
+        msg!("Error: Omnibus token address does not match derivation");
+        return Err(ProgramError::InvalidArgument);
+    }
+
+    let omnibus_token_account_signer_seeds: &[&[_]] = &[
+        &mint_info.key.to_bytes(),
+        br"confidential_transfer_omnibus",
+        &[omnibus_token_bump_seed],
+    ];
+
+    create_pda_account(
+        funder_info,
+        &Rent::get()?,
+        Account::get_packed_len(),
+        &id(),
+        system_program_info,
+        omnibus_info,
+        omnibus_token_account_signer_seeds,
+    )?;
+
+    Processor::process(
+        &id(),
+        &[omnibus_info.clone(), mint_info.clone()],
+        &crate::instruction::initialize_account3(
+            &id(),
+            omnibus_info.key,
+            mint_info.key,
+            omnibus_info.key,
+        )?
+        .data,
+    )
+}
+
+/// Processes an [UpdateAuditor] instruction.
+fn process_update_auditor(
+    accounts: &[AccountInfo],
+    new_auditor: &ConfidentialTransferAuditor,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let mint_info = next_account_info(account_info_iter)?;
+
+    check_program_account(mint_info.owner)?;
+    let mint_data = &mut mint_info.data.borrow_mut();
+    let mut mint = StateWithExtensionsMut::<Mint>::unpack(mint_data)?;
+    let auditor = mint.get_extension_mut::<ConfidentialTransferAuditor>()?;
+
+    let authority_info = next_account_info(account_info_iter)?;
+
+    Processor::validate_owner(
+        &id(),
+        &auditor.authority,
+        authority_info,
+        account_info_iter.as_slice(),
+    )?;
+
+    *auditor = *new_auditor;
+    Ok(())
+}
+
+/// Processes an [ApproveAccount] instruction.
+fn process_approve_account(accounts: &[AccountInfo]) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let account_to_approve_info = next_account_info(account_info_iter)?;
+    let mint_info = next_account_info(account_info_iter)?;
+
+    check_program_account(account_to_approve_info.owner)?;
+    let account_to_approve_data = &mut account_to_approve_info.data.borrow_mut();
```

Comment:
Having to hold a reference to the account data to keep rustc happy is annoying.  This is the only major wart I've found so far with the extension stuff!